src/Homework 5/Program.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+
+namespace ATM_Managment
+{
+    public class ATM
+    {
+        public delegate void Action(decimal balance, string action);
+        public event Action BalanceHandler;
+
+        private decimal _balance = 0;
+
+        public void AddFunds(decimal _count)
+        {
+            _balance += _count;
+            BalanceHandler?.Invoke(_balance, "deposite");
+        }
+        public void WithdrawFunds(decimal _count)
+        {
+            if (_count > _balance)
+            {
+                Console.WriteLine("You do not have enough funds to withdraw");
```

Comment:
You can add another event to report error for example, so that class that handles an event will write to console

Reply:
It is needed to be showen by you)